plugin.video.cbc-sports/addon.py

```diff
@@ -0,0 +1,495 @@
+	    return
+	stream = (re.compile('src="(.+?)"').findall(str(contents))[0])
+	if mp4base not in stream:
+		#stream = mp4base + stream
+		stream = stream
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports Live Schedule Playback stream: ' + str(stream), xbmc.LOGINFO)
+	if valid == 0:
+	    li = xbmcgui.ListItem(name, path=stream)
+	    li.setArt({'thumb': defaultimage, 'icon': defaultimage})
+	    xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, li)
+	else:
+	    iframe_return()
+
+#99
+def play(url):
+	item = xbmcgui.ListItem(path=url)
+	return xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, item)
+
+def iframe_return():			#  iframe return unavailable URL                
+	listitem = xbmcgui.ListItem(name, path=unavailableurl)
+	listitem.setArt({'thumb': defaultimage, 'icon': defaultimage})
+	xbmcplugin.setResolvedUrl(int(sys.argv[1]), False, listitem)
+
+def striphtml(data):
+	p = re.compile(r'<.*?>')
+	return p.sub('', data)
+
+
+def sanitize(data):
+	output = ''
+	for i in data:
+		for current in i:
+			if ((current >= '\x20') and (current <= '\xD7FF')) or ((current >= '\xE000') and (current <= '\xFFFD')) or ((current >= '\x10000') and (current <= '\x10FFFF')):
+			   output = output + current
+	return output
+
+
+
+def get_html(url):
+	req = urllib.request.Request(url)
+	req.add_header('User-Agent','User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux i686; rv:44.0) Gecko/20100101 Firefox/44.0')
+
+	try:
+		response = urllib.request.urlopen(req)
+		code = response.getcode()
+		cbclog = int(xbmcaddon.Addon().getSetting('cbclog'))
+		if cbclog >= 1:
+			xbmc.log('CBC Sports get_html code: ' + str(code), xbmc.LOGINFO)
+		if code == 403:              
+			xbmcgui.Dialog().notification(name, translation(30001), defaultimage, notetime, False)
+			html = 'error'
+			#sys.exit()
+		elif code == 22:              
+			xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+			html = 'error'
+			#sys.exit()	    
+		html = response.read()
+		response.close()
+	except urllib.error.URLError:              
+		xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+		html = 'error'
+		#sys.exit()	 
+	return html
+
+
+def get_params():
+	param = []
+	paramstring = sys.argv[2]
+	if len(paramstring) >= 2:
+		params = sys.argv[2]
+		cleanedparams = params.replace('?', '')
+		if (params[len(params) - 1] == '/'):
+			params = params[0:len(params) - 2]
+		pairsofparams = cleanedparams.split('&')
+		param = {}
+		for i in range(len(pairsofparams)):
+			splitparams = {}
+			splitparams = pairsofparams[i].split('=')
+			if (len(splitparams)) == 2:
+				param[splitparams[0]] = splitparams[1]
+
+	return param
+
+
+def addDir(name, url, mode, iconimage, fanart=False, infoLabels=True):
+	u = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + str(mode) + "&name=" + urllib.parse.quote_plus(name) + "&iconimage=" + urllib.parse.quote_plus(iconimage)
+	ok = True
+	liz = xbmcgui.ListItem(name)
```

Comment:
you'd get a bit of performance by adding `offscreen=False` in the listitem initializer

Reply:
Updated as suggested